src/language/schema/materializer.js

```diff
@@ -0,0 +1,263 @@
+/**
```

Comment:
I wish this file was @flow, I think it would clarify what the various `def` variables are.